tools/src/test/java/org/apache/kafka/tools/consumer/group/ResetConsumerGroupOffsetTest.java

```diff
@@ -62,506 +91,786 @@
-        BufferedWriter bw = new BufferedWriter(new FileWriter(file));
-        bw.write(consumerGroupCommand.exportOffsetsToCsv(exportedOffsets));
-        bw.close();
-        Map<TopicPartition, Long> exp1 = new HashMap<>();
-        exp1.put(t1p0, 2L);
-        exp1.put(t1p1, 2L);
-        Map<TopicPartition, Long> exp2 = new HashMap<>();
-        exp2.put(t2p0, 2L);
-        exp2.put(t2p1, 2L);
-
-        assertEquals(exp1, toOffsetMap(exportedOffsets.get(group1)));
-        assertEquals(exp2, toOffsetMap(exportedOffsets.get(group2)));
-
-        // Multiple --group's offset import
-        String[] cgcArgsExec = buildArgsForGroups(Arrays.asList(group1, group2), "--all-topics", "--from-file", file.getCanonicalPath(), "--dry-run");
-        ConsumerGroupCommand.ConsumerGroupService consumerGroupCommandExec = getConsumerGroupService(cgcArgsExec);
-        Map<String, Map<TopicPartition, OffsetAndMetadata>> importedOffsets = consumerGroupCommandExec.resetOffsets();
-        assertEquals(exp1, toOffsetMap(importedOffsets.get(group1)));
-        assertEquals(exp2, toOffsetMap(importedOffsets.get(group2)));
-
-        // Single --group offset import using "group,topic,partition,offset" csv format
-        String[] cgcArgsExec2 = buildArgsForGroup(group1, "--all-topics", "--from-file", file.getCanonicalPath(), "--dry-run");
-        ConsumerGroupCommand.ConsumerGroupService consumerGroupCommandExec2 = getConsumerGroupService(cgcArgsExec2);
-        Map<String, Map<TopicPartition, OffsetAndMetadata>> importedOffsets2 = consumerGroupCommandExec2.resetOffsets();
-        assertEquals(exp1, toOffsetMap(importedOffsets2.get(group1)));
-
-        adminZkClient().deleteTopic(TOPIC);
-    }
-
-    @Test
-    public void testResetWithUnrecognizedNewConsumerOption() {
-        String[] cgcArgs = new String[]{"--new-consumer", "--bootstrap-server", bootstrapServers(listenerName()), "--reset-offsets",
-            "--group", GROUP, "--all-topics", "--to-offset", "2", "--export"};
-        assertThrows(OptionException.class, () -> getConsumerGroupService(cgcArgs));
+    private Map<TopicPartition, Long> partitionToOffsets(String topic,
+                                                         Map<TopicPartition, OffsetAndMetadata> partitionInfo) {
+        return partitionInfo.entrySet()
+                .stream()
+                .filter(entry -> Objects.equals(entry.getKey().topic(), topic))
+                .collect(toMap(Map.Entry::getKey, e -> e.getValue().offset()));
     }
 
-    private void produceMessages(String topic, int numMessages) {
-        List<ProducerRecord<byte[], byte[]>> records = IntStream.range(0, numMessages)
-            .mapToObj(i -> new ProducerRecord<byte[], byte[]>(topic, new byte[100 * 1000]))
-            .collect(Collectors.toList());
-        kafka.utils.TestUtils.produceMessages(servers(), seq(records), 1);
+    private static List<String> generateIds(String name) {
+        return IntStream.rangeClosed(1, 3)
```

Comment:
Could we reduce the number? There are 5 (cluster) * 3 (groups) * 3 (topics) * 2 (protocols) = 90 ...